### PR TITLE
Google Drive storage

### PR DESCRIPTION
**Pull request by joraff**
Requires install of gdrive client https://github.com/prasmussen/gdrive

### CODE REVIEW

**Comment by joraff** [comment]
Happy to also do a PR to the website, wherever that repo is.

File: lib/backup/storage/google_drive.rb [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        @path           ||= 'backups'
+        path.sub!(/^\//, '')
+
+        required = %w{ refresh_token }
+        raise Error, "Configuration Error: a refresh_token is required" if refresh_token.nil?
+
+        raise Error, "Configuration Error: gdrive executable is required." if gdrive_exe.nil?
+      end
+
+      # private
```
Comment (tombruijn) [inline lib/backup/storage/google_drive.rb]: can be uncommented or removed?

**Comment by tombruijn** [comment]
Hi @joraff. Thanks for the pr!

I'm just not sure if I agree with the approach here. Using a third party tool, executable for uploading is not a strategy we've used before and I fear it might be a bit confusing for the end user as they actively have to install something separately. 

Plus, if we would use it, it should be registered as a utility in backup rather than specifying a path as a configure option on the storage itself. http://backup.github.io/backup/v4/utilities/

**Comment by joraff** [comment]
Hi @tombruijn - I had originally tried using google's api gem, and would have much preferred that myself, too. But they have dependencies on a large number of common gems at very different versions. Rather than play the dep res game over and over, I came across this utility and decided to give it a go.

I was unaware of the Utilities class and agree that's a much better place.

Is using a third party tool to upload much different than using a third party tool for backup generation? It's expected that if you want to backup a MySQL database that you have the mysql-cli tools installed. In a way you're using third-party code via gems (i.e. fog) to do many of the other upload functions - in this case it's just packaged differently.

**Comment by joraff** [comment]
@tombruijn I'm revising this and would like to know if the third party tool approach would even be considered, if implemented/integrated well?

**Comment by tombruijn** [comment]
Hi @joraff, no I don't think I'll merge this particular solution. I'll close the PR as well. If you have a different solution that doesn't rely on a third party tool for uploading to google drive I'd be happy to review it

**Pull request closed without merging**